### ISSUE TITLE
Build Issue with Latest code in Rust 

### ISSUE THREAD

**Issue by harshakhmk** [qdrant/qdrant#3158]
## Current Behavior

While I am building Qdrant as a standalone application in Ubuntu 20.04 , I am facing build issue

Error: failed to on run custom build command for api v1. 6.1 
Caused by :
Process didn't exit successfully (exit status: 1) 
Cargo:rerun if changed= src/grpc/proto/qdrant.proto
cargo:rerun-if-changed=src/grpc/proto
Error: Custom { Kind: Other, error: "Protoc failed: collections.proto:7:12 : explicit optional labels are disallowed in the proto 3 syntax. To define optional fields in the proto 3 simply remove the optional label as fields are optional by default.

## Steps to Reproduce

1.

I have installed [Rust toolchain ] (https://rustup.rs/) and followed the steps.
2.Installed few other dependencies on Ubuntu:
```
apt install build-essential libclang-dev protobuf-compiler
```

3.  ```
cargo build --release
```


<!--- Please make sure to include the data which could be used to reproduce the problem -->


## Expected Behavior
<!--- Tell us what should happen -->

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->

## Context (Environment)
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

<!--- Provide a general summary of the issue in the Title above -->

## Detailed Description
<!--- Provide a detailed description of the change or addition you are proposing -->

## Possible Implementation
<!--- Not obligatory, but suggest an idea for implementing addition or change -->

**Comment by harshakhmk** [comment]
I just followed the steps mentioned to build Protoc from source and updated its version [here](https://github.com/qdrant/qdrant/pull/1576/files#diff-d9657425e036fb0ad2912c2c24ea793648eb86de5b40aa0844276dd08ed01d87)

Now I am getting SIGSEGV: Invalid memory reference #2819 
I am using rustc: 1.75
Protoc version 22.2

**Comment by timvisee** [comment]
> I just followed the steps mentioned to build Protoc from source and updated its version [here](https://github.com/qdrant/qdrant/pull/1576/files#diff-d9657425e036fb0ad2912c2c24ea793648eb86de5b40aa0844276dd08ed01d87)
> 
> Now I am getting SIGSEGV: Invalid memory reference #2819 I am using rustc: 1.75 Protoc version 22.2

Do you see the same when using `protobuf-compiler` shipped by your OS? Because then I cannot reproduce this issue.

My details:
- Ubuntu 23.10
- Rust 1.75
- Qdrant `master` and `dev`

Two more questions:
- Are you sure you're not running Rust 1.73? Please check `cargo --version`.
- What version of Qdrant are you trying to build, are you using some specific branch?

**Comment by harshakhmk** [comment]
Yes when the followed the steps mentioned in the [here],(https://github.com/qdrant/qdrant/pull/1576/files#diff-d9657425e036fb0ad2912c2c24ea793648eb86de5b40aa0844276dd08ed01d87) I am getting SIGSEGV Invalid memory issue
1) Yes I am sure, rustc version is 1.75 and cargo version is also the same
2) I am using latest master branch

**Comment by generall** [comment]
Docker build passes, so the default assumption is that something is wrong with env
